test/smtcallback.js

```diff
@@ -211,15 +212,22 @@ tape('SMTCheckerCallback', function (t) {
       // `pragma experimental SMTChecker;` was deprecated in 0.8.4
       if (semver.gt(solc.semver(), '0.8.3')) {
         const engine = test.engine !== undefined ? test.engine : 'all';
-        settings = { modelChecker: { engine: engine } };
+        settings = {
+          modelChecker: {
+            engine: engine,
+            solvers: [
+              'smtlib2'
```

Comment:
Why is this test forcing smtlib2?

Reply:
to not use the z3 embedded in soljson, and rather use a local solver via the callback